src/main/kotlin/net/atos/zac/signalering/SignaleringenJob.kt

```diff
@@ -0,0 +1,377 @@
+            """.trimMargin()
+        )
+    }
+
+    /**
+     * Send the E-Mail warnings about the einddatum gepland
+     *
+     * @return the number of E-Mails sent
+     */
+    fun zaakEinddatumGeplandVerzenden(zaaktype: ZaakType, venster: Int): Int {
+        val verzonden = IntArray(1)
+        zoekenService.zoek(getZaakSignaleringTeVerzendenZoekParameters(DatumVeld.ZAAK_STREEFDATUM, zaaktype, venster))
+            .items.stream()
+            .map { zaakZoekObject: ZoekObject? -> zaakZoekObject as ZaakZoekObject? }
+            .map { zaakZoekObject: ZaakZoekObject? ->
+                buildZaakSignalering(
+                    getZaakSignaleringTarget(zaakZoekObject, SignaleringDetail.STREEFDATUM),
+                    zaakZoekObject,
+                    SignaleringDetail.STREEFDATUM
+                )
+            }
+            .filter { obj: Signalering? -> Objects.nonNull(obj) }
+            .forEach { signalering: Signalering? -> verzonden[0] += verzendZaakSignalering(signalering) }
+        return verzonden[0]
+    }
+
+    /**
+     * Send the E-Mail warnings about the uiterlijke einddatum afdoening
+     *
+     * @return the number of E-Mails sent
+     */
+    fun zaakUiterlijkeEinddatumAfdoeningVerzenden(
+        zaaktype: ZaakType,
+        venster: Int
+    ): Int {
+        val verzonden = IntArray(1)
+        zoekenService.zoek(
+            getZaakSignaleringTeVerzendenZoekParameters(
+                DatumVeld.ZAAK_FATALE_DATUM,
+                zaaktype,
+                venster
+            )
+        )
+            .items.stream()
+            .map { zaakZoekObject: ZoekObject? -> zaakZoekObject as ZaakZoekObject? }
+            .map { zaakZoekObject: ZaakZoekObject? ->
+                buildZaakSignalering(
+                    getZaakSignaleringTarget(zaakZoekObject, SignaleringDetail.FATALE_DATUM),
+                    zaakZoekObject,
+                    SignaleringDetail.FATALE_DATUM
+                )
+            }
+            .filter { obj: Signalering? -> Objects.nonNull(obj) }
+            .forEach { signalering: Signalering? -> verzonden[0] += verzendZaakSignalering(signalering) }
+        return verzonden[0]
+    }
+
+    fun getZaakSignaleringTarget(zaak: ZaakZoekObject?, detail: SignaleringDetail): String? {
+        if (signaleringenService.readInstellingenUser(
+                SignaleringType.Type.ZAAK_VERLOPEND,
+                zaak!!.behandelaarGebruikersnaam
+            ).isMail &&
+            !signaleringenService.findSignaleringVerzonden(
+                getZaakSignaleringVerzondenParameters(
+                    zaak.behandelaarGebruikersnaam, zaak.uuid,
+                    detail
+                )
+            ).isPresent
+        ) {
+            return zaak.behandelaarGebruikersnaam
+        }
+        return null
+    }
+
+    fun buildZaakSignalering(
+        target: String?,
+        zaakZoekObject: ZaakZoekObject?,
```

Comment:
```suggestion
        zaakZoekObject: ZaakZoekObject,
```

? it looks from the code below that this field should not be nullable at all. Doesn't seem to make sense to make this nullable in any case right?

Reply:
But I see below that you use `zaakZoekObject!!`. This means that zaakZoekObject simply cannot be null or else this would throw a nullpointer. I don't see a call to `zoekenService.zoek` at all here? So I would at least in this function just make it non-nullable by removing the `?` in the parameter.

Reply:
Normally in Kotlin having to resort to `!!` really means that there is another issue you need to resolve. It is really hardly ever needed.